interfaces/console/src/components/SiteCard/index.tsx

```diff
@@ -21,56 +17,67 @@ import {
   Menu,
   MenuItem,
   Skeleton,
+  Tooltip,
   Typography,
+  useTheme,
 } from '@mui/material';
-import { useRouter } from 'next/navigation';
 import React, { useState } from 'react';
 
 interface SiteCardProps {
   siteId: string;
   name: string;
   address: string;
   userCount?: number;
-  connectionStatus?: string;
-  batteryStatus?: string;
-  signalStrength?: string;
+  siteUptimeSeconds?: number | null;
+  batteryPercentage?: number | null;
+  backhaulSpeed?: number | null;
   loading?: boolean;
   handleSiteNameUpdate: (siteId: string, newSiteName: string) => void;
+  maxAddressLength?: number;
 }
 
+const truncateText = (text: string, maxLength: number): string => {
+  if (text.length <= maxLength) return text;
+  return `${text.substring(0, maxLength)}...`;
+};
+
 const SiteCard: React.FC<SiteCardProps> = ({
   siteId,
   name,
   address,
-  connectionStatus = 'Online',
-  batteryStatus = 'Charged',
-  signalStrength = 'Strong',
+  userCount = 0,
+  siteUptimeSeconds,
+  batteryPercentage,
+  backhaulSpeed,
   handleSiteNameUpdate,
-  loading = false,
+  loading,
+  maxAddressLength = 49,
 }) => {
-  const router = useRouter();
-
   const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+
+  const displayAddress = loading ? '' : truncateText(address, maxAddressLength);
+
   const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
     event.stopPropagation();
     setAnchorEl(event.currentTarget);
   };
-  const handleMenuClick = (action: string) => {
-    if (handleSiteNameUpdate) {
-      handleSiteNameUpdate(siteId, name);
-    }
-  };
 
   const handleClose = () => {
     setAnchorEl(null);
   };
+
+  const handleMenuClick = () => {
+    handleSiteNameUpdate(siteId, name);
+    handleClose();
+  };
+
   const navigateToDetails = () => {
-    router.push(`/console/sites/${siteId}`);
+    window.location.href = `/console/sites/${siteId}`;
```

Comment:
Is `router.push` is not working as expecting? why we're adopting `window.location.href`